### PR TITLE
@W-16336722: [MSDK Android] QR Code Log In Using External QR Code Reader Does Not Display Main Activity

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🥁 _*Ready For Review!*_ 🎸

This is a companion to https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2625

### CODE REVIEW

File: AndroidNativeKotlinTemplate/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -20,6 +20,16 @@
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+
+            <!-- Uncomment when enabling log in via Salesforce UI Bridge API generated QR codes -->
+            <!-- This declares the app's support for QR code login deep links.  The scheme can be customized by the app to match any scheme used in the QR Code's URL -->
+<!--            <intent-filter>-->
```
Comment (JohnsonEricAtSalesforce) [inline AndroidNativeKotlinTemplate/app/src/main/AndroidManifest.xml:26]: Note this intent filter is attached to the app's main activity now, which was the crux of fixing this issue.

File: AndroidNativeKotlinTemplate/app/src/main/java/com/salesforce/androidnativekotlintemplate/MainActivity.kt [hunk 2]
```diff
@@ -156,4 +163,16 @@ class MainActivity : SalesforceActivity() {
...
         })
     }
-}
\ No newline at end of file
+
+    // region QR Code Login Via Salesforce Identity API UI Bridge Public Implementation
+
+    private fun useQrCodeLoginUrl() {
+        val uiBridgeApiParameters = uiBridgeApiParametersFromQrCodeLoginUrl(intent.data.toString())
+        SalesforceSDKManager.getInstance().apply {
```
Comment (JohnsonEricAtSalesforce) [inline AndroidNativeKotlinTemplate/app/src/main/java/com/salesforce/androidnativekotlintemplate/MainActivity.kt:171]: Here's the app's opportunity to provide the front door URL before it soon gets the REST client, which will automatically display the login activity as a side effect.

Are there any other paths to displaying the login activity we might need to intercept? Once these parameters are set, the login activity will use them one time.

**Review by JohnsonEricAtSalesforce — COMMENTED** [review]
Here's my self review and code walk-through to start the discussion.  Thanks!

File: AndroidNativeKotlinTemplate/app/src/main/java/com/salesforce/androidnativekotlintemplate/MainActivity.kt (hunk 2 above)
Comment (wmathurin) [reply AndroidNativeKotlinTemplate/app/src/main/java/com/salesforce/androidnativekotlintemplate/MainActivity.kt:171]: >Are there any other paths to displaying the login activity we might need to intercept? Once these parameters are set, the login activity will use them one time.

I don't think so.

**Review by wmathurin — APPROVED** [review]

**Pull request merged**